skimage/draw/_random_shapes.py

```diff
@@ -312,16 +344,28 @@ def random_shapes(image_shape,
                     msg = 'Intensity range must lie within (0, 255) interval'
                     raise ValueError(msg)
 
+    if background is None:
+        background = [255] * num_channels
+    elif isinstance(background, int):
+        background = [background]
+    if len(background) != num_channels:
+        raise ValueError('Nr of background values must match nr of channels')
+    for intensity in background:
```

Comment:
It seems that operations on `background` take too many lines. Let's just cast it to ndarray from the beginning, and use the vectorized ops.